File: archive/v/vb.net/reverse-string.vb
Public Module ReverseString
  Public Sub Main(args() As String)
    Dim input = String.Empty
    If args IsNot Nothing Then
      input = String.Join(" ", args)
    End If

    If String.IsNullOrEmpty(input) Then
      System.Console.Write("Please provide some text to reverse: ")
      input = System.Console.ReadLine()
    End If

    System.Console.WriteLine($"Input: {input}")

    Dim reversedString = ReverseString(input)
    System.Console.WriteLine($"Reversed: {reversedString}")
  End Sub

  Public Function ReverseString(input As String) As String
    Dim chars = input.ToCharArray()

    System.Console.WriteLine(chars.Length)

    Dim reversedChars(input.Length) As Char

    For i = 0 To chars.Length - 1
      reversedChars(input.Length - 1 - i) = chars(i)
    Next

    Return New String(reversedChars)
  End Function
End Module

File: archive/c/c-sharp/ReverseString.cs
if (args is [var input] && !string.IsNullOrEmpty(input))
    Console.WriteLine(Reverse(input.AsSpan()));

static string Reverse(ReadOnlySpan<char> s)
{
    int n = s.Length;
    char[] result = new char[n];

    for (int i = 0; i < n; i++)
        result[i] = s[n - 1 - i];

    return new string(result);
}